pkg/queue/queue.go

```diff
@@ -1,36 +1,25 @@
 package queue
 
 import (
-	"encoding/json"
-	"github.com/webhookx-io/webhookx/utils"
+	"context"
 	"time"
 )
 
-type TaskMessage struct {
-	ID string
-
-	data []byte
-	Data interface{}
-}
-
-func NewTaskMessage(data interface{}) *TaskMessage {
-	task := &TaskMessage{
-		ID:   utils.UUID(),
-		Data: data,
-	}
-	return task
-}
-
-func (t *TaskMessage) String() string {
-	return t.ID + ":" + string(t.data)
+type Message struct {
+	ID          string
+	Data        []byte
+	Time        time.Time
+	WorkspaceID string
 }
 
-func (t *TaskMessage) UnmarshalData(v interface{}) error {
-	return json.Unmarshal(t.data, v)
+type Options struct {
+	Count   int64
+	Block   bool
+	Timeout time.Duration
 }
 
-type TaskQueue interface {
-	Add(task *TaskMessage, scheduleAt time.Time) error
-	Get() (task *TaskMessage, err error)
-	Delete(task *TaskMessage) error
+type Queue interface {
+	Enqueue(ctx context.Context, message *Message) error
+	Dequeue(ctx context.Context, opts *Options) ([]*Message, error)
+	Delete(ctx context.Context, message []*Message) error
```

Comment:
Add the `ctx`  to `Message` for tracing.

Reply:
I feel this could be done when the tracing package is ready.